src/components/Box/index.tsx

```diff
@@ -27,8 +28,8 @@ export type BoxProps = {
   children?: ReactNode
   height?: number | string
   width?: number | string
-} & XStyledProps &
-  HTMLAttributes<HTMLElement> & {
+} & (HTMLAttributes<HTMLElement> | Omit<AllHTMLAttributes<HTMLElement>, 'as'>) &
```

Comment:
I don't like the Omit workaround, as should be typed as React.Element which is union of FunctionComponent, Component and JSX.Element no ?